test/cleanup.go

```diff
@@ -24,6 +24,26 @@ import (
 	"os/signal"
 )
 
+// MultipleCleanupOnInterrup will execute the multiple functions to cleanup if an interrupt signal is caught
+func MultipleCleanupOnInterrup(cleanupCh chan func()) {
```

Comment:
Could we instead adjust the semantics of `CleanupOnInterrupt` to build a stack of functions to run on interrupt instead?

Not sure if that suffices, but would a globally managed slice be good enough? Or a `sync.Map` for good measure, to allow concurrent access. We'd store all the functions in there and register one listener (`sync.Once` is your friend here) for a signal and run all the cleanup functions there.

Or we keep the code more or less as-is, but have a global `WaitGroup`. We run each cleanup by adding `1` to the WaitGroup and launch another goroutine that does `os.Exit(1)`, once the WaitGroup finishes.

WDYT?

Reply:
Yes, I agree. I think using `sync.Map` to store all functions and call once is better. Updated.

Reply:
We could also make the function variadic while we're at it so you can combine multiple calls.